ldms/src/ldmsd/ldmsd_row_cache.c

```diff
@@ -262,6 +301,41 @@ int ldmsd_row_cache(ldmsd_row_cache_t rcache,
 
 	pthread_mutex_lock(&rcache->lock);
 
+	rc = clock_gettime(CLOCK_REALTIME, &ts);
+	if (rc) {
+		rc = errno;
+		goto out;
+	}
+
+	if (rcache->cfg_timeout.tv_sec == 0 && rcache->cfg_timeout.tv_nsec == 0)
+		goto skip_cleanup;
+
+	/* process buckets */
+	count = 3;
```

Comment:
This should be sizeof(rcache->group_bucket). We have this _magic_ constant 3 running around this logic.